src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java

```diff
@@ -529,4 +553,354 @@ public static void main(String[] args) {
+    Map<String, String> parameters = new HashMap<>();
+    /* cell formatting string */
+    parameters.put("alignCellCenter", " align=center");
+    /* Useful words and phrases */
+    parameters.put("phraseMultipleEntriesAllowed", I18N.getText("phrase.multipleEntriesAllowed"));
+    parameters.put("mapVisionDistance", I18N.getText("sight.default.distance"));
+
+    /* Build list of useful words and phrases */
+    List<String> helpKeys = I18N.getMatchingKeys(Pattern.compile("^word."));
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^option.type.")));
+    /* Shape names */
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^shape.type.name.")));
+    /* example text */
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^sight.example.")));
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^light.example.")));
+
+    /* Generate parameter map from list */
+    for (String key : helpKeys) {
+      parameters.putIfAbsent(
+          uncapitalize(capitalize(key, '.').replace(".", "")), I18N.getText(key));
+    }
+
+    /* Parameterised strings - need to be done individually */
+    parameters.put(
+        "wikiLinkReferral",
+        I18N.getText(
+            "sightLight.wikiLinkReferral",
+            "<i>wiki.rptools.info/index.php/Introduction_to_Lights_and_Sights</i>"));
+    if (MapTool.getLanguage().toLowerCase().startsWith("en")) {
+      /* remove translated version of words for English locales. */
+      parameters.put(
+          "optionDescriptionShape",
+          I18N.getText("sightLight.optionDescription.shape", "", "", "", "", "", "")
+              .replace("(", "")
+              .replace(")", ""));
+    } else {
+      parameters.put(
+          "optionDescriptionShape",
+          I18N.getText(
+              "sightLight.optionDescription.shape",
+              parameters.get("shapeTypeNameBeam"),
+              parameters.get("shapeTypeNameCircle"),
+              parameters.get("shapeTypeNameCone"),
+              parameters.get("shapeTypeNameGrid"),
+              parameters.get("shapeTypeNameHexagon"),
+              parameters.get("shapeTypeNameSquare")));
+    }
+    parameters.put(
+        "optionDescriptionPersonalSightComponentColor",
+        I18N.getText("sightLight.optionDescription.personalSight.component.color", "#rrggbb"));
+    parameters.put(
+        "optionDescriptionRestriction",
+        I18N.getText("sightLight.optionDescription.restriction", "gm", "owner"));
+
+    /* everything else */
+    helpKeys = I18N.getMatchingKeys("sightLight");
+    for (String key : helpKeys) {
+      parameters.putIfAbsent(
+          uncapitalize(
+              capitalize(key, new char[] {'.'}).replace("SightLight", "").replace(".", "")),
+          I18N.getText(key));
+    }
+    return parameters;
+  }
+
+  /**
+   * Creates HTML for both sight and light help
+   *
+   * @return String[]
+   */
+  private String[] generateHelpText() {
```

Comment:
Maybe extract this into a helper class? The method length would suggest this and linters probably already complain about it.

Reply:
Bwahahaha... you should have seen the long version.